packages/sdk/src/conversation/middleware.ts

```diff
@@ -120,33 +132,44 @@ export class CommandResponseMiddleware implements Middleware {
   }
 
   private classifyActivity(activity: Activity): ActivityType {
-    if (this.isCommandReceived(activity)) {
-      return ActivityType.CommandReceived;
+    if (activity.type === ActivityTypes.Message) {
+      return ActivityType.CurrentBotMessaged;
     }
 
     return ActivityType.Unknown;
   }
 
-  private isCommandReceived(activity: Activity): boolean {
-    if (this.commandHandlers) {
-      const commandText = this.getActivityText(activity);
-      const handlers = this.filterCommandHandler(commandText, this.commandHandlers);
-      return handlers.length > 0;
+  private matchPattern(pattern: string | RegExp, text: string): boolean | RegExpMatchArray {
+    const normalizedText = text?.trim().toLocaleLowerCase();
+
+    if (typeof pattern === "string") {
+      const test = new RegExp(pattern as string, "i");
+      if (normalizedText && normalizedText.match(test)) {
+        return true;
+      }
+    } else if (pattern instanceof RegExp) {
+      const test = pattern as RegExp;
+      if (normalizedText && normalizedText.match(test)) {
+        const matches = normalizedText.match(test);
```

Comment:
Is it possible to only call `match` once?

Reply:
updated.